### DISCUSSION TITLE
Clone Multiple Transactions at Once

### DISCUSSION THREAD

**Discussion by YamGZZ** [firefly-iii/firefly-iii#5766] [General]
Hi,

There are multiple ways to accomplish my specific scenario so I will break this down into separate questions to hopefully avoid any confusion. I am essentially trying to create a work around for the piggy bank feature to allow a negative balance by using savings accounts. 

1. I'm interested in a way to clone multiple transactions at one time. Currently, after I import transactions there are items I manually apply a tag to. I then search for all transactions that include that tag and click through each one to clone them. After they're all cloned, I go through and select the second copy and apply another tag to identify it as the copy. Then I use rules to change the copied transaction source account. 

1a. Is it possible to make cloned transactions keep their original transaction date rather than re-assign them to the present date?
1b. Are there plans to implement shift + click capabilities to select transactions between a certain range shown on the page? 

2. Is there (or will there?) be a way to export transactions that have a certain tag between a specified date? this way I can re-import a narrowed list into a different account. From my understanding of the export feature in the documentation, it creates separate files for all of the different flags. I haven't experimented with this too much yet but I plan on doing so to make this process less clicky. 

As a new Linux user I've really enjoyed the process of setting up this whole system and experimenting with its capabilities. 
Thanks!

**Comment by JC5** [comment]
Thanks for the compliments! Glad to hear you like Firefly III. 

I'm curious to hear what you try to achieve and why you believe this is part of the solution. 

https://en.wikipedia.org/wiki/XY_problem

**Reply by YamGZZ** [reply]
At the moment, my partner and I each have our own "piggy banks"/fun money in our main savings accounts. We each get $100 a month to spend on anything. What ever amount is left over at the end of the month gets rolled over to the next month and $100 is added to that amount. We don't mind if some months we spend more than what we have in our piggy bank and go into the negative amount as we have enough in the savings account to be able to pull from. That negative amount would just be added to the next months $100. 

I figured instead of using piggy banks in Firefly, I could just use the savings accounts so we can go into the negative. Then I can also setup recurring transfers to "transfer" money from the savings accounts to the piggy banks. We also frequently have multiple transactions that we'd like to categorize but also add a tag to so we can exclude from reports. I usually clone the transactions that are tagged as fun money paid from the savings account and duplicate them to piggy bank savings account to keep track. Coming from using excel to keep track of our budget, I didn't realize how hard it would be to find a solution that would support something like this. But it makes sense. 

Edit: Another Idea I just thought of that may be a solution to my problem would be to enable rules to not only modify, but create transactions.

**Reply by JC5** [reply]
Thanks for the explanation. I'm not sure if I can solve these problems for you in Firefly III. I don't really understand why you copy transactions around, if it's other than to support the solution for the monthly allowance. I don't know if I can really help with that.

- You could use budgets to manage the fun money; they do [auto-rollover](https://docs.firefly-iii.org/firefly-iii/concepts/budgets/#rollover-add-an-amount-every-period) using the amount left or reset to 100. You won't be able to carry over negative amounts though.
- Cloning will always make the transaction jump to today.

> Coming from using excel to keep track of our budget, I didn't realize how hard it would be to find a solution that would support something like this.

That has to do with the fact that your solution is so unique. You seem to go through all these lenghts to exclude these transactions from your reports when it seems a lot easier to give them a budget and just ignore them when you see the report. If you separate transactions in budgets you can track them separately easily.

**Reply by YamGZZ** [reply]
That makes sense. I'll keep experimenting and maybe use a mix between firefly and Excel to make the process just a little more automated. It's not too bad right now. Thanks!
